pyproject.toml

```diff
@@ -1,56 +1,103 @@
-[tool.poetry]
+[project]
 name = "slowapi"
-version = "0.1.9"
-description = "A rate limiting extension for Starlette and Fastapi"
-authors = ["Laurent Savaete <laurent@where.tf>"]
-license = "MIT"
-
+version = "0.2.0"
+description = "A rate limiting extension for Starlette and FastAPI"
 readme = "README.md"
+authors = [
+    { name = "Laurent Savaete", email = "laurent@where.tf"},
+    { name = "Noelle Wang", email = "73260931+No767@users.noreply.github.com" }
+]
+requires-python = ">=3.10,<4"
+dependencies = [
+    "fastapi>=0.116.1,<1",
+    "limits>=5.5.0,<6",
+]
+
+[project.optional-dependencies]
+redis = [
+    "redis[hiredis]>=6.3.0,<7",
+]
+valkey = [
+    "valkey[libvalkey]>=6.1.0,<7",
+]
+
+[dependency-groups]
+dev = [
+    "ruff>=0.12.7,<1",
+    "pyright[nodejs]>=1.1.403,<2",
+    "lefthook>=1.12.2,<2",
+    {include-group = "docs"},
+    {include-group = "test"}
+]
+
+docs = [
+    "mkdocs>=1.6.1,<2",
+    "mkdocstrings[python]>=0.30.0,<1"
+]
+test = [
+    "pytest>=8.4.1,<9",
+    "pytest-asyncio>=1.1.0,<2",
+    "coverage>=7.10.2,<8",
+    "httpx>=0.28.1,<1",
+    "hiro>=1.1.1,<2"
+]
+
+[tool.pyright]
+include = ["slowapi",  "tests"]
+exclude = ["**/__pycache__", "docs",]
+reportMissingImports = "error"
+typeCheckingMode = "standard"
+reportUnnecessaryTypeIgnoreComment = "warning"
 
-repository = "https://github.com/laurents/slowapi"
-homepage = "https://github.com/laurents/slowapi"
-documentation = "https://slowapi.readthedocs.io/en/latest/"
-
-include = ["slowapi/py.typed"]
```

Comment:
I haven't looked at whether this is still needed, but flagging it here so we don't forget.